apps/voting/contracts/Voting.sol

```diff
@@ -200,7 +291,29 @@ contract Voting is IForwarder, AragonApp {
     function vote(uint256 _voteId, bool _supports, bool _executesIfDecided_deprecated) external voteExists(_voteId) {
         require(_canVote(_voteId, msg.sender), ERROR_CAN_NOT_VOTE);
         require(!_supports || _getVotePhase(votes[_voteId]) == VotePhase.Main, ERROR_CAN_NOT_VOTE);
-        _vote(_voteId, _supports, msg.sender);
+        _vote(_voteId, _supports, msg.sender, false);
+    }
+
+    function voteFor(uint256 _voteId, bool _supports, address _voteFor) external voteExists(_voteId) {
+        require(_canVoteFor(msg.sender, _voteFor), ERROR_CAN_NOT_VOTE_FOR);
+        require(_canVote(_voteId, _voteFor), ERROR_CAN_NOT_VOTE);
+        require(!_supports || _getVotePhase(votes[_voteId]) == VotePhase.Main, ERROR_CAN_NOT_VOTE);
+        _vote(_voteId, _supports, _voteFor, true);
+    }
+
+    /**
+     * TODO: Update the voteForMultiple function to allow voting with the entire voting power
+     * of both self-managed address and managed addresses / write voteWithFullPower function
+     */
+    function voteForMultiple(uint256 _voteId, bool _supports, address[] _voteForList) external voteExists(_voteId) {
+        require(!_supports || _getVotePhase(votes[_voteId]) == VotePhase.Main, ERROR_CAN_NOT_VOTE);
```

Comment:
I'd propose extracting this check into a separate fn since it's non-trivial and is used in few places